qmetaobject/src/qtdeclarative.rs

```diff
@@ -975,14 +973,15 @@ mod qjsvalue_tests {
 ///
 /// ```
 /// # extern crate qmetaobject; use qmetaobject::*;
+/// # use std::ffi::CStr;
 /// #[derive(Default, QObject)]
 /// struct QExampleQmlPlugin {
 ///     base: qt_base_class!(trait QQmlExtensionPlugin),
 ///     plugin: qt_plugin!("org.qt-project.Qt.QQmlExtensionInterface/1.0"),
 /// }
 ///
 /// impl QQmlExtensionPlugin for QExampleQmlPlugin {
-///     fn register_types(&mut self, uri: &std::ffi::CStr) {
+///     fn register_types(&mut self, uri: &CStr) {
```

Comment:
I think it's useful in the example to see what exactly CStr is (that it is std::ffi::CStr and not some other things)

Reply:
Let me see.  I expect docs generator to be smart enough to link it to std::ffi::Cstr.

Reply:
No, unfortunately, the are no clickable words in example blocks whatsoever. WTF, rust? Next things next, let me check what is the official position of std docs regarding this point.

Reply:
Looking at [std::ffi](https://doc.rust-lang.org/std/ffi/index.html) docs, they are explicitly and without hiding behind hashtags `#` *using* their items in each and every example. And they don't bother hiding things imported from other modules, as can be seen in e.g. [std::time::SystemTime](https://doc.rust-lang.org/std/time/struct.SystemTime.html).

Maybe we should try too.

Reply:
The general rule of de-facto seems to be like this:

1. Show your imports explicitly,
2. One crate per line, use braces to group items,
3. Std is an exception to the rule 2: its second level modules may be written one per line, as if they were separate crates.
4. Alphabetic ordering, std has no privileges to be on top of others.
5. No grouping of imports, no blank lines between std/libs/workspace/`crate`/`self` etc. That kind of stuff IntelliJ-Rust absolutely **loves** doing while optimizing imports.

Reply:
Also, `# extern crate qmetaobject;` is not needed at all, since rust doctest inserts it automatically anyway, and `#[macro_use] extern crate qmetaobject;` is no longer needed, since we switched to 2018 edition a while ago.

Reply:
Some links to back up my claims:

- https://doc.rust-lang.org/rustdoc/documentation-tests.html#pre-processing-examples rules regarding inserting `extern crate`
- https://docs.rs/async-compat/0.2.1/async_compat/struct.Compat.html (sorted uses)
- https://docs.rs/bimap/0.6.1/bimap/hash/struct.BiHashMap.html#method.with_hashers (std on top, but otherwise is fine)